lib/ruby_routes/router/builder.rb

```
@@ -0,0 +1,71 @@
+            raise ArgumentError, "Disallowed DSL method: #{method_name.inspect}"
+          end
+
+          # (Optional) shallow dup args to avoid later mutation side‑effects
+          safe_args = arguments.map { |a| a.is_a?(String) || a.is_a?(Numeric) || a.is_a?(Symbol) ? a : a.dup rescue a }
+
+          router.public_send(method_name, *safe_args, &definition_block)
+        end
+
+        router.finalize!
+        router
+      end
```

RuboCop (1.76.1)

[convention] Cyclomatic complexity for `build` is too high. [8/7]

(Metrics/CyclomaticComplexity)